react-native-cli/index.js

```diff
@@ -203,33 +203,46 @@ function createProject(name, verbose, rnPackage) {
 
 function getInstallPackage(rnPackage) {
   var packageToInstall = 'react-native';
-  var valideSemver = semver.valid(rnPackage);
-  if (valideSemver) {
-    packageToInstall += '@' + valideSemver;
+  var isValidSemver = semver.valid(rnPackage);
+  if (isValidSemver) {
+    packageToInstall += '@' + isValidSemver;
   } else if (rnPackage) {
     // for tar.gz or alternative paths
     packageToInstall = rnPackage;
   }
   return packageToInstall;
 }
 
+function isYarnInstalled(boolCallback) {
+  exec('yarn --version', function (error) {
+    boolCallback(!e);
+  });
+}
+
 function run(root, projectName, rnPackage) {
-  exec('npm install --save --save-exact ' + getInstallPackage(rnPackage), function(e, stdout, stderr) {
-    if (e) {
-      console.log(stdout);
-      console.error(stderr);
-      console.error('`npm install --save --save-exact react-native` failed');
-      process.exit(1);
+  isYarnInstalled(function (hasYarn) {
+    var installCommand;
+    if (hasYarn) {
+      installCommand = 'yarn add ' + getInstallPackage(rnPackage) + ' --exact';
```

Comment:
`getInstallPackage()` can return "react-native" so better to keep `--exact` I think?